src/main/integrations/net-breadcrumbs.ts

```diff
@@ -0,0 +1,189 @@
+ */
+function parseOptions(optionsIn: ClientRequestConstructorOptions | string): { method: string; url: string } {
+  const { method, options } =
+    typeof optionsIn === 'string'
+      ? { method: 'GET', options: urlModule.parse(optionsIn) }
+      : { method: (optionsIn.method || 'GET').toUpperCase(), options: optionsIn };
+
+  let url = 'url' in options ? options.url : undefined;
+
+  if (!url) {
+    const urlObj: urlModule.UrlObject = {};
+    urlObj.protocol = options.protocol || 'http:';
+
+    if (options.host) {
+      urlObj.host = options.host;
+    } else {
+      if (options.hostname) {
+        urlObj.hostname = options.hostname;
+      } else {
+        urlObj.hostname = 'localhost';
+      }
+
+      if (options.port) {
+        urlObj.port = options.port;
+      }
+    }
+
+    const pathObj = urlModule.parse(options.path || '/');
+    urlObj.pathname = pathObj.pathname;
+    urlObj.search = pathObj.search;
+    urlObj.hash = pathObj.hash;
+    url = urlModule.format(urlObj);
+  }
+
+  return {
+    method,
+    url,
+  };
+}
+
+type RequestOptions = string | ClientRequestConstructorOptions;
+type RequestMethod = (opt: RequestOptions) => ClientRequest;
+type WrappedRequestMethodFactory = (original: RequestMethod) => RequestMethod;
+
+/** */
+function createWrappedRequestFactory(
+  breadcrumbsEnabled: boolean,
+  tracingEnabled: boolean,
+): WrappedRequestMethodFactory {
+  return function wrappedRequestMethodFactory(originalRequestMethod: RequestMethod): RequestMethod {
+    return function requestMethod(this: typeof net, options: RequestOptions): ClientRequest {
+      // eslint-disable-next-line @typescript-eslint/no-this-alias
+      const netModule = this;
+
+      const { url, method } = parseOptions(options);
+      const request = originalRequestMethod.apply(netModule, [options]) as ClientRequest;
+
+      if (url.match(/sentry_key/) || request.getHeader('x-sentry-auth')) {
+        return request;
+      }
+
+      let span: Span | undefined;
+
+      const scope = getCurrentHub().getScope();
+      if (scope && tracingEnabled) {
+        const parentSpan = scope.getSpan();
+
+        if (parentSpan) {
+          span = parentSpan.startChild({
+            description: `${method} ${url}`,
+            op: 'request',
+          });
+
+          request.setHeader('sentry-trace', span.toTraceparent());
+        }
+      }
+
+      return request
+        .once('response', function (this: ClientRequest, res: IncomingMessage): void {
+          // eslint-disable-next-line @typescript-eslint/no-this-alias
+          const req = this;
+          if (breadcrumbsEnabled) {
+            addRequestBreadcrumb('response', method, url, req, res);
+          }
+          if (tracingEnabled && span) {
+            if (res.statusCode) {
+              span.setHttpStatus(res.statusCode);
+            }
+            span.finish();
+          }
+        })
+        .once('error', function (this: ClientRequest, _error: Error): void {
```

Comment:
we don't need to `captureException` here?

Reply:
I'm guessing we want to leave it up to the `net.request` user whether they want to capture failures as errors or not?